framed-messages/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+pub mod framed;
+
+use bytes::BytesMut;
+use futures_util::{SinkExt, TryStreamExt};
+use iroh::{
+    endpoint::Connection,
+    protocol::{AcceptError, ProtocolHandler},
+};
+use serde::{Deserialize, Serialize};
+
+use crate::framed::FramedBiStream;
+
+/// Each protocol is identified by its ALPN string.
+///
+/// The ALPN, or application-layer protocol negotiation, is exchanged in the connection handshake,
+/// and the connection is aborted unless both nodes pass the same bytestring.
+pub const ALPN: &[u8] = b"iroh/examples/messages/0";
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Move {
+    pub from: (u8, u8),
```

Comment:
this is not how you denote chess movements..

Reply:
Also... this is just for demonstration. I'll add a comment and hope people don't hate b5/me too much for this.

Reply:
And yes... you can use SAN, but no I won't do that :P